fractal_graphing/algos/algos.go

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2016 Pawel Rozlach.
+// License: https://creativecommons.org/licenses/by-nc-sa/4.0/
+
+// Based on work by Alan A. A. Donovan & Brian W. Kernighan
+// which can be found at:
+// https://github.com/adonovan/gopl.io.git
+
+// Mandelbrot emits a PNG image of the Mandelbrot fractal.
+package algos
+
+import "math/cmplx"
+
+//(sur) Is it a problem that I make func type private and at the same time
+// map using it public ?
+type AlgoFunc func(r, i float64) (uint8, uint8)
+
+// (sur) Is there a way to do _getattr_ in a simple way ? Or is the "map"
+// approach idiomatic enough ?
+var Str2funcMapping = map[string]AlgoFunc{
+	"newton":         Newton,
+	"acos":           Acos,
+	"mandelbrotC64":  MandelbrotC64,
+	"mandelbrotC128": MandelbrotC128,
+	"sqrt":           Sqrt,
+}
+
+// Mandelbrot algo, but with complex128 type resolution
+func MandelbrotC128(r, i float64) (uint8, uint8) {
+	var z complex128 = complex(r, i)
+	const iterations = 20000
```

Comment:
Even though it's used only once and could be easily removed, I decided that it would be nice to name important params somehow, make them less of a "magic number" and more of a "this is the number of iterations of the algorithm".